docarray/array/doc_list/doc_list.py

```diff
@@ -267,8 +267,10 @@ def validate(
     ):
         from docarray.array.doc_vec.doc_vec import DocVec
 
-        if isinstance(value, (cls, DocVec)):
+        if isinstance(value, cls):
             return value
+        elif isinstance(value, DocVec):
+            return value.to_doc_list()
```

Comment:
here we need to test if the documen_type of the value and self are compatible.

If they are you can silence mypy by return  `cast(T, )`